src/averageAges.js

```diff
@@ -55,7 +65,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child => (
+      people.find(mother => (
+        child.mother === mother.name)) && child.sex === 'm'))
+    : people.filter(child => (
+      people.find(mother => (
+        child.mother === mother.name))));
+
+  return children.reduce((sum, child) => {
+    return sum + child.born - people.find(mother => (
+      child.mother === mother.name)).born;
```

Comment:
Move `people.find(mother => (
      child.mother === mother.name)).born;` to separate variable